### PR TITLE
Improve Slack Sanitization

### PR DESCRIPTION
**Pull request by vigneshp826**
**Proposed changes**:
- Find mailto or http links like <mailto:xyz@rasa.com|xyz@rasa.com> or '<http://url.com|url.com> in 
   text generated by slack and substitute it with original content

**Status (please check what you already did)**:
- [X] made PR ready for code review
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog
- [ ] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa_nlu#code-style) for instructions)

### CODE REVIEW

**Comment by vigneshp826** [comment]
@erohmensing not able to get much out of this build error log 

"Makefile:40: recipe for target 'lint' failed
make: *** [lint] Error 1"

Can you please help me :)?

**Comment by msamogh** [comment]
@vigneshp826 You have to reformat your files using black.

**Comment by msamogh** [comment]
But thank you for your PR, we'll give it a review as soon as possible.

**Comment by erohmensing** [comment]
A changelog entry would be good too please :)

**Comment by vigneshp826** [comment]
changelog entry under [Unreleased 1.4.0] - master ?

**Comment by msamogh** [comment]
@vigneshp826 Yes

**Comment by vigneshp826** [comment]
added changelog entry

File: rasa/core/channels/slack.py [hunk 1]
```diff
@@ -199,12 +195,24 @@ def _sanitize_user_message(text, uids_to_remove):
...
+
+        pattern = r"\<(mailto:|(http|https):\/\/).*\|.*\>"
+        match = re.search(pattern, text)
+
+        if match:
+            regex = match.group(0)
+            replacement = regex.split("|")[1]
+            replacement = replacement.replace('>', '')
+            text = text.replace(regex, replacement)
+        else:
```
Comment (msamogh) [inline rasa/core/channels/slack.py]: The `else` part can be dropped.

**Comment by msamogh** [comment]
But it looks good otherwise. Thanks for your contribution!

File: rasa/core/channels/slack.py [hunk 2]
```diff
@@ -180,16 +180,12 @@ def _is_user_message(slack_event):
     @staticmethod
     def _sanitize_user_message(text, uids_to_remove):
         """Remove superfluous/wrong/problematic tokens from a message.
-
```
Comment (msamogh) [inline rasa/core/channels/slack.py]: Could you also add these empty lines back?

Comment (vigneshp826) [reply rasa/core/channels/slack.py]: hope this is fine now :)

**Pull request merged**